client/src/components/menu-bar/MenuBar.js

```diff
@@ -0,0 +1,107 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import AppBar from '@material-ui/core/AppBar';
+import IconButton from '@material-ui/core/IconButton';
+import Typography from '@material-ui/core/Typography';
+import MenuItem from '@material-ui/core/MenuItem';
+import Menu from '@material-ui/core/Menu';
+import Toolbar from '@material-ui/core/Toolbar';
+import AccountCircle from '@material-ui/icons/AccountCircle';
+import Avatar from '@material-ui/core/Avatar';
+import auth from '../../auth/auth';
+import config from '../../config/globalConfig';
+
+export default class MenuBar extends Component {
+  state = {
+    loggedIn: false,
+    user: {},
+    anchorEl: null,
+  };
+
+  componentDidMount() {
+    this.fetchUserIfLoggedIn();
+  }
+
+  componentDidUpdate() {
+    this.fetchUserIfLoggedIn();
+  }
+
+  fetchUserIfLoggedIn = () => {
```

Comment:
says fetchUserIfLoggedIn
1) Does not return a user
2) has an if (!loggedIn.... as the main branch which is confusing

Reply:
1. it updates user details in components state
2. yes, confusing name "fetchUserIfLoggedIn" changing it.